File: SDKs/amazonaws.com/glue/2017-03-31/go/pkg/models/operations/batchcreatepartition.go
package operations

import (
	"openapi/pkg/models/shared"
)

type BatchCreatePartitionXAmzTargetEnum string

const (
	BatchCreatePartitionXAmzTargetEnumAwsGlueBatchCreatePartition BatchCreatePartitionXAmzTargetEnum = "AWSGlue.BatchCreatePartition"
)

type BatchCreatePartitionHeaders struct {
	XAmzAlgorithm     *string                            `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                            `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                            `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                            `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                            `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                            `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                            `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        BatchCreatePartitionXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type BatchCreatePartitionRequest struct {
	Headers BatchCreatePartitionHeaders
	Request shared.BatchCreatePartitionRequest `request:"mediaType=application/json"`
}

type BatchCreatePartitionResponse struct {
	AlreadyExistsException               *interface{}
	BatchCreatePartitionResponse         *shared.BatchCreatePartitionResponse
	ContentType                          string
	EntityNotFoundException              *interface{}
	GlueEncryptionException              *interface{}
	InternalServiceException             *interface{}
	InvalidInputException                *interface{}
	OperationTimeoutException            *interface{}
	ResourceNumberLimitExceededException *interface{}
	StatusCode                           int64
}

File: SDKs/amazonaws.com/glue/2017-03-31/typescript/src/sdk/models/operations/batchcreatepartition.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum BatchCreatePartitionXAmzTargetEnum {
    AwsGlueBatchCreatePartition = "AWSGlue.BatchCreatePartition"
}


export class BatchCreatePartitionHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: BatchCreatePartitionXAmzTargetEnum;
}


export class BatchCreatePartitionRequest extends SpeakeasyBase {
  @Metadata()
  headers: BatchCreatePartitionHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.BatchCreatePartitionRequest;
}


export class BatchCreatePartitionResponse extends SpeakeasyBase {
  @Metadata()
  alreadyExistsException?: any;

  @Metadata()
  batchCreatePartitionResponse?: shared.BatchCreatePartitionResponse;

  @Metadata()
  contentType: string;

  @Metadata()
  entityNotFoundException?: any;

  @Metadata()
  glueEncryptionException?: any;

  @Metadata()
  internalServiceException?: any;

  @Metadata()
  invalidInputException?: any;

  @Metadata()
  operationTimeoutException?: any;

  @Metadata()
  resourceNumberLimitExceededException?: any;

  @Metadata()
  statusCode: number;
}
